packages/react-celo/src/connectors/wallet-connect.ts

```diff
@@ -45,16 +45,25 @@ export default class WalletConnectConnector
 
   // this is called automatically and is what gives us the uri for the qr code to be scanned
   async initialise(): Promise<this> {
+    if (this.initialised) {
+      return this;
+    }
     const wallet = this.kit.getWallet() as WalletConnectWallet;
-
-    wallet.on('session_update', this.onSessionCreated);
     wallet.on('session_update', this.onSessionUpdated);
```

Comment:
having onSessionCreate listen to session_update caused issues. instead onSessionUpdated now handles session_update all the time and will emit WalletConnected. 

I believe there isnt any danger in emitting an on Connected event when wallet is already connected as i just updates values in a reducer and if they are the same nothing happens anyway